### ISSUE TITLE
Run Issue

### ISSUE THREAD

**Issue by mustii82** [haxpor/aseprite-macos-buildsh#1]
On run I get this error:

third_party/externals/icu            @ ec9c1133693148470ffe2e5e53576998e3650c1d
Done. Made 43 targets from 26 files in 26ms
ninja: Entering directory `out/Release'
ninja: no work to do.
Note: checking out 'v1.2.10-beta4-47-g4dd7febb9'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 4dd7febb9 lua: Add support to specify ExportSpriteSheet params
/Users/praktikant/Desktop/aseprite-macos-buildsh-for-aseprite-1.2.9/aseprite.sh: line 148: cmake: command not found
ninja: error: loading 'build.ninja': No such file or directory

**Comment by haxpor** [comment]
Thanks for reporting. I checked, and possibly it's about the parsing of skia branch. I push back the fix soon.

**Comment by mustii82** [comment]
I replaced the source folder Wirth 1.2.9 which can't be build as well :/

**Comment by haxpor** [comment]
Thanks for feedback and test it out. I think similarly. Have time to take a look now, I'll make building latest version works first. For specifying version to build against, less likely to happen as build steps for each version changed dramatically. Anyway, we will see.

**Comment by haxpor** [comment]
**Update**: I tried building against skia's branches of `aseprite-m67` and `aseprite-m71` against latest commit in master branch, or release tag `v1.2.9`, all didn't work either compilation of skia or aseprite itself. Related to https://github.com/aseprite/aseprite/issues/1981 for latest build on macOS mojave. I'll update more.

**Update 2**: Building against older version like 1.2.9 is not possible anymore on mojave as build requirement ups, and break stuff. Possibly need to wait for upstream update.

**What you can try**
If you're on older version macOS like 10.13, then you can try to use https://github.com/haxpor/aseprite-macos-buildsh/releases/tag/for-aseprite-1.2.9 then modify source as follows

* Replace https://github.com/haxpor/aseprite-macos-buildsh/blob/for-aseprite-1.2.9/aseprite.sh#L126 with `SKIA_BRANCH=aseprite-m67`
* Replace https://github.com/haxpor/aseprite-macos-buildsh/blob/for-aseprite-1.2.9/aseprite.sh#L147 with `git checkout v1.2.9`

**Comment by haxpor** [comment]
Update, I'm able to successfully build it now. In short, if you can't get pass the following error

```
Undefined symbols for architecture x86_64:
  "is_orientation_marker(unsigned char const*, unsigned long, SkEncodedOrigin*)", referenced from:
      SkWebpCodec::MakeFromStream(std::__1::unique_ptr<SkStream, std::__1::default_delete<SkStream> >, SkCodec::Result*) in libskia.a(webp.SkWebpCodec.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

Try disable webp option via `skia_use_libwebp=false`.

I'm in the process to release a fixed version of broken parsing branch name from remote source, and unable to complete compile which is this issue).

**Comment by mustii82** [comment]
I get this Message on the newest Release:

WARNING at the command-line "--args":1:160: Build argument has no effect.
is_official_build=true skia_use_system_expat=false skia_use_system_icu=false skia_use_libjpeg_turbo=false skia_use_system_libpng=false skia_use_system_libwebp=false skia_use_system_zlib=false skia_use_libwebp=false extra_cflags_cc=["-frtti"]
                                                                                                                                                               ^----
The variable "skia_use_system_libwebp" was set as a build argument
but never appeared in a declare_args() block in any buildfile.

To view all possible args, run "gn args --list <out_dir>"

The build continued as if that argument was unspecified.

Done. Made 36 targets from 22 files in 25ms
ninja: Entering directory `out/Release'
[973/973] link libskia.a
Note: checking out 'v1.2.10-beta4-73-gb3eba1c95'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at b3eba1c95 Fix unnecessary warning saving TGA (fix #1830)
aseprite.sh: line 149: cmake: command not found

**Comment by haxpor** [comment]
The warning at the top is expected, just ignore and it will be fine at the end.
The bottom-most, you need `cmake` installed on your machine. You can either build from source, or download a binary, see this [page](https://cmake.org/install/).
